lib/FreeRTOS/freertos-plus-mqtt/freertos_mqtt_agent.c

```diff
@@ -1627,207 +1441,156 @@ MQTTStatus_t MQTTAgent_ResumeSession( MQTTContextHandle_t mqttContextHandle,
-                                uint32_t timeoutMs,
-                                bool * pSessionPresent )
-{
-    MQTTStatus_t statusReturn;
-
-    if( ( mqttContextHandle < MQTT_AGENT_MAX_SIMULTANEOUS_CONNECTIONS ) &&
-        ( mqttContexts[ mqttContextHandle ].connectStatus == MQTTNotConnected ) )
-    {
-        statusReturn = MQTT_Connect( &( mqttContexts[ mqttContextHandle ] ),
-                                     pConnectInfo,
-                                     pWillInfo,
-                                     timeoutMs,
-                                     pSessionPresent );
-    }
-    else
-    {
-        statusReturn = MQTTIllegalState;
-    }
-
-    return statusReturn;
-}
-
-/*-----------------------------------------------------------*/
-/*_RB_ Should return MQTTStatus_t. */
-MQTTStatus_t MQTTAgent_Subscribe( MQTTContextHandle_t mqttContextHandle,
-                                  MQTTSubscribeInfo_t * pSubscriptionInfo,
-                                  IncomingPublishCallback_t incomingPublishCallback,
-                                  void * incomingPublishCallbackContext,
-                                  CommandCallback_t commandCompleteCallback,
-                                  void * commandCompleteCallbackContext,
-                                  uint32_t blockTimeMS )
+MQTTStatus_t MQTTAgent_Subscribe( MQTTAgentContext_t * pMqttAgentContext,
+                                  MQTTAgentSubscribeArgs_t * pSubscriptionArgs,
+                                  CommandInfo_t * pCommandInfo )
 {
     MQTTStatus_t statusReturn;
 
     statusReturn = createAndAddCommand( SUBSCRIBE,                      /* commandType */
-                                        mqttContextHandle,              /* mqttContextHandle */
-                                        pSubscriptionInfo,              /* pMqttInfoParam */
-                                        commandCompleteCallback,        /* commandCompleteCallback */
-                                        commandCompleteCallbackContext, /* pCommandCompleteCallbackContext */
-                                        incomingPublishCallback,        /* incomingPublishCallback */
-                                        incomingPublishCallbackContext, /* pIncomingPublishCallbackContext */
-                                        blockTimeMS );
+                                        pMqttAgentContext,              /* mqttContextHandle */
+                                        pSubscriptionArgs,              /* pMqttInfoParam */
+                                        pCommandInfo->cmdCompleteCallback,        /* commandCompleteCallback */
+                                        pCommandInfo->pCmdCompleteCallbackContext, /* pCommandCompleteCallbackContext */
+                                        NULL,        /* incomingPublishCallback */
+                                        NULL, /* pIncomingPublishCallbackContext */
```

Comment:
Are these 2 arguments required with the new design? Is this something that is planning to be removed?

Reply:
They're not required, I just didn't clean up these internal functions right now

Reply:
OK. Sounds good.